### ISSUE TITLE
Question about performance reproduction

### ISSUE THREAD

**Issue by yuyq18** [facebookresearch/RPG_KDD2025#9]
Hi authors,

Thank you for your impressive work and for sharing the code!

I am writing to inquire about the reproduction of performance results. I have **strictly followed the "Reproduction Protocol"** described in [Issue #3](https://github.com/facebookresearch/RPG_KDD2025/issues/3) and the **README** instructions to re-run the entire pipeline from scratch:

1.  **Data Preparation:** I used the **OpenAI API** to generate text representations and used FAISS to compute semantic IDs, exactly as described.
2.  **Training & Inference:** I executed the exact scripts provided in the `README.md` for each dataset without modification.

### Reproduction Scripts
I used the following commands for training, as specified in the repository:

```bash
# Sports_and_Outdoors
CUDA_VISIBLE_DEVICES=0 python main.py \
    --category=Sports_and_Outdoors \
    --lr=0.003 \
    --temperature=0.03 \
    --n_codebook=16 \
    --num_beams=100 \
    --n_edges=30 \
    --propagation_steps=5
    
CUDA_VISIBLE_DEVICES=0 python main.py \
    --category=Beauty \
    --lr=0.01 \
    --temperature=0.03 \
    --n_codebook=32 \
    --num_beams=20 \
    --n_edges=200 \
    --propagation_steps=3
    
CUDA_VISIBLE_DEVICES=0 python main.py \
    --category=Toys_and_Games \
    --lr=0.003 \
    --temperature=0.03 \
    --n_codebook=16 \
    --num_beams=200 \
    --n_edges=20 \
    --propagation_steps=3
    
```

My reproduction was conducted on an NVIDIA A100 (40GB).

### Results Comparison

I evaluated the model under two settings:

- **Full sorting:** To estimate the performance upper bound, consistent with the validation setup.
- **Graph-based decoding:** Using the default parameters from the repository scripts above.

While the results on Sports are reasonably close, I observed a **significant performance drop on Beauty and Toys**. Specifically, the result on **Toys** is nearly half of the reported performance and falls significantly below the TIGER baseline.

**Full Results Table:**

| **Dataset** | **Metric** | **Tiger (Paper)** | **RPG (Paper)** | **My Repro (Full sorting)** | **My Repro (Graph)** |
| ----------- | ---------- | ----------------- | --------------- | --------------------------- | -------------------------- |
| **Sports**  | R@5        | 0.0264            | 0.0314          | 0.0295                      | 0.0291                     |
|             | N@5        | 0.0181            | 0.0216          | 0.0205                      | 0.0202                     |
|             | R@10       | 0.0400            | 0.0463          | 0.0448                      | 0.0437                     |
|             | N@10       | 0.0225            | 0.0263          | 0.0254                      | 0.0249                     |
| **Beauty**  | R@5        | 0.0454            | 0.0550          | 0.0443                      | 0.0424                     |
|             | N@5        | 0.0321            | 0.0381          | 0.0315                      | 0.0303                     |
|             | R@10       | 0.0648            | 0.0809          | 0.0595                      | 0.0572                     |
|             | N@10       | 0.0384            | 0.0464          | 0.0364                      | 0.0350                     |
| **Toys**    | R@5        | 0.0521            | 0.0592          | 0.0300                      | 0.0287                     |
|             | N@5        | 0.0371            | 0.0401          | 0.0229                      | 0.0220                     |
|             | R@10       | 0.0712            | 0.0869          | 0.0401                      | 0.0381                     |
|             | N@10       | 0.0432            | 0.0490          | 0.0261                      | 0.0250                     |

Could you kindly provide some insights into what might be causing this discrepancy? Since I adhered strictly to the protocol (including OpenAI embeddings and the exact commands from the README), I am puzzled by the large gap, especially on the Toys dataset.

Any advice would be greatly appreciated. Thanks!

**Comment by hyp1231** [comment]
Thank you for the detailed reproduction report. Below are a few checks that may help narrow down the issue.

1. First isolate whether the gap comes from embeddings. Could you please try rerunning using `sentence-t5-base` embeddings (as in Table 4)?
    * If sentence-t5 results match ours, please refer to section 2 below.
    * If sentence-t5 still cannot match, please directly refer to section 3.
2. We suggest spot-checking the OpenAI embeddings, e.g., whether any embeddings become all-zero vectors (likely due to network issues).
3. As discussed in the paper and in #3, the best hyperparameters are unlikely to be the same across different GPUs and environments. Our released configs are the best in our own environment. Specifically, we tunethe  learning rate $\in$ {0.01, 0.003, 0.001} and temperature $\tau \in$ {0.03, 0.05, 0.07}. We select the hyperparameter combination that performs the best on the validation set, and report the corresponding test set performance.

**Comment by hyp1231** [comment]
We've also uploaded a copy of the obtained OpenAI embeddings in #8. You may use them to:
1. directly train your models, or
2. cross-validate against your own embeddings (e.g., check whether each item's embedding vector matches ours).

**Comment by yuyq18** [comment]
Hi @hyp1231,

Thank you so much for the prompt response and for uploading the embeddings in #8!

Following your suggestions, I re-ran the experiments using the OpenAI embeddings you provided. I am happy to report that the performance gap has been resolved, and I was able to **successfully reproduce the results** reported in the paper. It appears the discrepancy was indeed caused by the embeddings generated on my side.

Here are the reproduced results (using your embeddings with **Full Sorting** inference):

| Dataset | Metric | Tiger (Paper) | RPG (Paper) | My Repro (Full sorting) |
| :--- | :--- | :--- | :--- | :--- |
| **Sports** | R@5 | 0.0264 | 0.0314 | 0.0305 |
| | N@5 | 0.0181 | 0.0216 | 0.0214 |
| | R@10 | 0.0400 | 0.0463 | 0.0448 |
| | N@10 | 0.0225 | 0.0263 | 0.0260 |
| **Beauty** | R@5 | 0.0454 | 0.0550 | 0.0540 |
| | N@5 | 0.0321 | 0.0381 | 0.0372 |
| | R@10 | 0.0648 | 0.0809 | 0.0787 |
| | N@10 | 0.0384 | 0.0464 | 0.0452 |
| **Toys** | R@5 | 0.0521 | 0.0592 | 0.0581 |
| | N@5 | 0.0371 | 0.0401 | 0.0389 |
| | R@10 | 0.0712 | 0.0869 | 0.0864 |
| | N@10 | 0.0432 | 0.0490 | 0.0480 |

Additionally, regarding the hyperparameters, I would like to mention that I did not perform the grid search for learning rate and temperature as suggested. I **simply used the default parameters provided in the repository scripts**, and they were **sufficient to reproduce the results** closely as shown above.

Thanks again for your assistance! I will close this issue now.
